pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -327,3 +335,101 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 			true)
 	}
 }
+
+func (c *Controller) storeDenyConnection(pktIn *ofctrl.PacketIn) error {
+
+	packet, err := binding.ParsePacketIn(pktIn)
+	if err != nil {
+		return fmt.Errorf("error in parsing packetin: %v", err)
+	}
+
+	denyConn, err := c.generateDenyConn(pktIn, packet)
+	if err != nil {
+		return fmt.Errorf("error when getting deny connection information: %v", err)
+	}
+	c.denyConnectionStore.AddOrUpdateConnection(denyConn)
+	return nil
+}
+
+func (c *Controller) generateDenyConn(pktIn *ofctrl.PacketIn, packet *binding.Packet) (*flowexporter.DenyConnection, error) {
+	denyConn := flowexporter.DenyConnection{}
+
+	// Get 5-tuple information
+	flowKey := flowexporter.Tuple{
+		SourcePort:      packet.SourcePort,
+		DestinationPort: packet.DestinationPort,
+		Protocol:        packet.IPProto,
+	}
+	if packet.IsIPv6 {
+		flowKey.SourceAddress = packet.SourceIP.To16()
+		flowKey.DestinationAddress = packet.DestinationIP.To16()
+		denyConn.IsIPv6 = true
+	} else {
+		flowKey.SourceAddress = packet.SourceIP.To4()
+		flowKey.DestinationAddress = packet.DestinationIP.To4()
+	}
+	denyConn.FlowKey = flowKey
+
+	// Map local source IP and destination IP
+	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(flowKey.SourceAddress.String())
+	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(flowKey.DestinationAddress.String())
+	node, _ := env.GetNodeName()
+	if !srcFound && !dstFound {
+		klog.Warningf("Cannot map any of the IP %s or %s to a local Pod", flowKey.SourceAddress.String(), flowKey.DestinationAddress.String())
+	}
+	if srcFound && sIface.Type == interfacestore.ContainerInterface {
+		denyConn.SourcePodName = sIface.ContainerInterfaceConfig.PodName
+		denyConn.SourcePodNamespace = sIface.ContainerInterfaceConfig.PodNamespace
+		denyConn.SourceNodeName = node
+	}
+	if dstFound && dIface.Type == interfacestore.ContainerInterface {
+		denyConn.DestinationPodName = dIface.ContainerInterfaceConfig.PodName
+		denyConn.DestinationPodNamespace = dIface.ContainerInterfaceConfig.PodNamespace
+		denyConn.DestinationNodeName = node
+	}
+
+	matchers := pktIn.GetMatches()
+	var match *ofctrl.MatchField
+	// Get table ID
+	tableID := binding.TableIDType(pktIn.TableId)
+	// Get disposition Allow, Drop or Reject
+	match = getMatchRegField(matchers, uint32(openflow.DispositionMarkReg))
+	id, err := getInfoInReg(match, openflow.APDispositionMarkRange.ToNXRange())
+	if err != nil {
+		return nil, fmt.Errorf("error when getting disposition from reg: %v", err)
+	}
+	disposition := openflow.DispositionToString[id]
+
+	// For K8s network policy drop action, we cannot get name/namespace.
+	if tableID == openflow.IngressDefaultTable {
+		denyConn.IngressNetworkPolicyRuleAction = disposition
+	} else if tableID == openflow.EgressDefaultTable {
+		denyConn.EgressNetworkPolicyRuleAction = disposition
+	} else { // Get name and namespace for Antrea Network Policy
```

Comment:
This should be true for Antrea Cluster Network Policy too, right? If so, please add that here.

Reply:
Yes. Thanks for reminding.